pkg/api/v1alpha1/cluster_types.go

```diff
@@ -135,6 +136,20 @@ type ControlPlaneConfiguration struct {
 	Endpoint *Endpoint `json:"endpoint,omitempty"`
 	// MachineGroupRef defines the machine group configuration for the control plane.
 	MachineGroupRef *Ref `json:"machineGroupRef,omitempty"`
+	// Taints define the set of taints to be applied on control plane nodes
+	Taints []corev1.Taint `json:"taints,omitempty"`
+}
+
+func TaintsSliceEqual(s1, s2 []corev1.Taint) bool {
```

Comment:
if s1 and s2 have same taints with different order, we shall consider them same

Reply:
can we add unit test for this func?

Reply:
I will add it in the next pr.